modules/tinymce/src/core/main/ts/util/ArrUtils.ts

```diff
@@ -93,21 +104,20 @@ const indexOf = function (a, v) {
   return -1;
 };
 
-const reduce = function (collection, iteratee, accumulator?, thisArg?) {
-  let i = 0;
+const reduce: {
+  <T, R>(collection: ArrayLike<T>, iteratee: (acc: R, item: T, index: number) => R, accumulator: R, thisArg?: any): R;
+  <T>(collection: ArrayLike<T>, iteratee: (acc: T, item: T, index: number) => T, accumulator?: undefined, thisArg?: any): T;
```

Comment:
This special case shouldn't be necessary, as people can just use the two-argument version and supply the same argument twice?

Reply:
As discussed via slack, collapsing it would end up having the wrong `acc` type in specific cases, so this is a little more accurate and we'll keep it for now.